File: java/hash.java
// half hash program

import java.util.Scanner;

public class hash {

    public static int getNum() {

        Scanner scan = new Scanner(System.in);

        int x;
        System.out.print("Enter int for number of rows: ");
        x = scan.nextInt();

        return x;
    }

    public static void main(String[] args) {

        int rows = getNum();

        for (int x = rows; x >= 1; x--) {

            for (int y = 1; y <= x; y++) {
                System.out.print("#");
            }

            System.out.println();
        }
    }
}

File: cpp/think/hash.cpp
// think like a programmer
// hash program

#include <iostream>
using namespace std;

int getRows() {

    int rows;
    cout << "Enter integer for number of rows: ";
    cin >> rows;

    return rows;
}

int main() {

    int rows;
    rows = getRows();

    for (int x = rows; x >= 1; x--) {
        for(int y = 1; y <= x; y++) {
            cout << "#";
        }
        cout << endl;
    }

    return 0;
}

// END
